### PR TITLE
Feature/ingress frontend

### PR DESCRIPTION
**Pull request by chengyumeng**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

```release-note
Wayne supports users to create inress in the foreground.
Background management for ingress will be provided through other PR.
Also, some designs will be refactored recently.


```

### CODE REVIEW

**Comment by chengyumeng** [comment]
@BennieMeng  I will add i18n code in next PR.

**Comment by chengyumeng** [comment]
I have added admin page！

File: src/frontend/src/app/admin/ingress/ingress.component.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      this.retrieve();
+    }
+  }
+
+  openModal(): void {
+    this.createEdit.newOrEditIngress();
+  }
+
+  deleteIngress(ingress: Ingress) {
+    let deletionMessage = new ConfirmationMessage(
```
Comment (BennieMeng) [inline src/frontend/src/app/admin/ingress/ingress.component.ts]: 你不是有tslint了吗，这种后边不改的变量要用const

File: src/frontend/src/app/portal/ingress/ingress.module.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import {NgModule} from '@angular/core';
+import {IngressComponent} from './ingress.component';
```
Comment (BennieMeng) [inline src/frontend/src/app/portal/ingress/ingress.module.ts]: 这种不是都要加空格吗，括号前后

File: src/frontend/src/app/shared/client/v1/ingress.service.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    if (Object.keys(pageState.sort).length !== 0 && pageState.sort.by !== 'app.name') {
+      let sortType: any = pageState.sort.reverse ? `-${pageState.sort.by}` : pageState.sort.by;
+      params = params.set('sortby', sortType);
+    }
+    if ((typeof(appId) === 'undefined') || (!appId)) {
+      appId = '0';
+    }
+    return this.http
+      .get(`/api/v1/apps/${appId}/ingresses`, {params: params})
+
```
Comment (BennieMeng) [inline src/frontend/src/app/shared/client/v1/ingress.service.ts]: 这里是有空一行吗

File: src/frontend/src/app/admin/ingresstpl/create-edit-ingresstpl/create-edit-ingresstpl.component.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    this.currentForm.reset();
+  }
+
+  onSubmit() {
+    if (this.isSubmitOnGoing) {
+      return;
+    }
+    this.isSubmitOnGoing = true;
+    if (!this.aceBox.isValid) {
+      alert('语法有误，请检查！');
```
Comment (BennieMeng) [inline src/frontend/src/app/admin/ingresstpl/create-edit-ingresstpl/create-edit-ingresstpl.component.ts]: 这种最好用我们messageHandle,alert 有点不友好

**Pull request merged**